sdk/maps/Azure.ResourceManager.Maps/tests/ScenarioTests/MapsAccountTests.cs

```diff
@@ -0,0 +1,208 @@
+        {
+        }
+
+        private ResourceGroupResource ResourceGroup { get; set; }
+
+        private MapsAccountCollection MapCollection { get; set; }
+
+        private async Task SetCollection()
+        {
+            ResourceGroup = await CreateResourceGroupAsync();
+            MapCollection = ResourceGroup.GetMapsAccounts();
+        }
+
+        [Test]
+        public async Task MapsAccountCreateTest()
+        {
+            await SetCollection();
+
+            // prepare account properties
+            string accountName = Recording.GenerateAssetName("maps");
+            var parameters = GetDefaultMapsAccountData();
+
+            // Create account
+            var newAccount = (await MapCollection.CreateOrUpdateAsync(WaitUntil.Completed, accountName, parameters)).Value;
+            VerifyAccountProperties(newAccount.Data, true, Name.S0);
+
+            // now get the account
+            var account = (await MapCollection.GetAsync(accountName)).Value;
+            VerifyAccountProperties(account.Data, true, Name.S0);
+
+            // now delete the account
+            await account.DeleteAsync(WaitUntil.Completed);
+            var falseResult = (await MapCollection.ExistsAsync(accountName)).Value;
+            Assert.IsFalse(falseResult);
+        }
+
+        [Test]
+        public async Task MapsAccountUpdateTest()
+        {
+            await SetCollection();
+
+            // prepare account properties
+            string accountName = Recording.GenerateAssetName("maps");
+            var parameters = GetDefaultMapsAccountData();
+
+            // Create account
+            var newAccount = (await MapCollection.CreateOrUpdateAsync(WaitUntil.Completed, accountName, parameters)).Value;
+            VerifyAccountProperties(newAccount.Data, true, Name.S0);
+
+            // create new parameters which are almost the same, but have different tags
+            var newParameters = GetDefaultMapsAccountData();
+            newParameters.Tags.Clear();
+            newParameters.Tags.Add("key3", "value3");
+            newParameters.Tags.Add("key4", "value4");
+            newParameters.Sku.Name = Name.S1;
+            var updatedAccount = (await MapCollection.CreateOrUpdateAsync(WaitUntil.Completed, accountName, newParameters)).Value;
+            VerifyAccountProperties(updatedAccount.Data, false, skuName: Name.S1);
+            Assert.AreEqual(2, updatedAccount.Data.Tags.Count);
+            Assert.AreEqual("value3", updatedAccount.Data.Tags["key3"]);
+            Assert.AreEqual("value4", updatedAccount.Data.Tags["key4"]);
+        }
+
+        [Test]
+        public async Task MapsAccountDeleteTest()
+        {
+            await SetCollection();
+
+            // Delete an account which does not exist
+            var id = MapsAccountResource.CreateResourceIdentifier(DefaultSubscription.Data.SubscriptionId, ResourceGroup.Data.Name, "missingaccount");
+            var falseAccount = new MapsAccountResource(Client, id);
```

Comment:
We should not use internal methods in test cases.
```suggestion
            var falseAccount = Client.GetMapsAccountResource(id);
```

Reply:
Fixed